src/shogun/clustering/KMeansBase.cpp

```diff
@@ -130,6 +130,112 @@ void KMeansBase::compute_cluster_variances()
 	}
 }
 
+SGMatrix<float64_t> KMeansBase::get_stds() const
+{
+	SGMatrix<float64_t> stds(dimensions, k);
+
+	auto rhs = distance->get_lhs()->as<DenseFeatures<float64_t>>();
+	SGVector<int32_t> cluster_assignments;
+	SGVector<int64_t> weights_set;
+	std::tie(cluster_assignments, weights_set, std::ignore) =
+	    compute_cluster_assignments(k);
+
+	auto cluster_indexes = std::vector<SGVector<index_t>>(k);
+	auto cluster_counters = SGVector<index_t>(k);
+	for (int32_t current_cluster = 0; current_cluster < k; ++current_cluster)
+	{
+		cluster_indexes[current_cluster] =
+		    SGVector<index_t>(weights_set[current_cluster]);
+		cluster_counters[current_cluster] = 0;
+	}
+
+	for (int32_t i = 0; i < rhs->get_num_vectors(); ++i)
+	{
+		int point_cluster = cluster_assignments[i];
+		int& cluster_counter = cluster_counters[point_cluster];
+		cluster_indexes[point_cluster][cluster_counter] = i;
+		++cluster_counter;
+	}
+
+	for (int32_t i = 0; i < k; ++i)
+	{
+		stds.set_column(
+		    i, rhs->copy_subset(cluster_indexes[i])
```

Comment:
we don't want to copy things to compute std-dev. I think you could do this with a view, but might need some work to make the std call work

Reply:
Check View.h

Reply:
also maybe we can do the dynamic cast only once outside the loop?